core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -323,6 +323,58 @@ impl EthNamespace {
         Ok(tx_count?.map(|(_, count)| count))
     }
 
+    #[tracing::instrument(skip(self))]
+    pub async fn get_block_receipts_impl(
+        &self,
+        block_id: BlockId,
+    ) -> Result<Vec<TransactionReceipt>, Web3Error> {
+        const METHOD_NAME: &str = "get_block_receipts";
+
+        let method_latency = API_METRICS.start_block_call(METHOD_NAME, block_id);
+
+        let block = self
```

Comment:
You might want to add `self.state.start_info.ensure_not_pruned(block_id)?;` before getting the block to check that it's not pruned from the storage in order to get a more informative error. Compare with `get_block_impl()` method above.